integration/system/api_test.go

```diff
@@ -0,0 +1,24 @@
+package system
+
+import (
+	"net/http"
+	"testing"
+
+	"github.com/moby/moby/v2/internal/testutil/request"
+	"gotest.tools/v3/assert"
+	is "gotest.tools/v3/assert/cmp"
+)
+
+func TestAPIErrorNotFoundJSON(t *testing.T) {
+	ctx := setupTest(t)
+
+	// 404 is a different code path to normal errors, so test separately
+	httpResp, body, err := request.Get(ctx, "/notfound", request.JSON)
+	assert.NilError(t, err)
+	assert.Equal(t, httpResp.StatusCode, http.StatusNotFound)
+	assert.Assert(t, is.Contains(httpResp.Header.Get("Content-Type"), "application/json"))
+
+	b, err := request.ReadBody(body)
+	assert.NilError(t, err)
+	assert.Equal(t, getErrorMessage(t, b), "page not found")
```

Comment:
I opened a PR to add a utility;

- https://github.com/moby/moby/pull/51220

If that is merged, this could become something like;

```suggestion
	httpResp, _, err := request.Get(testutil.GetContext(c), "/notfound", request.JSON)
	assert.NilError(c, err)
	assert.Equal(c, httpResp.StatusCode, http.StatusNotFound)

	var respErr common.ErrorResponse
	assert.NilError(c, request.ReadJSONResponse(httpResp, &respErr))
	assert.Error(c, respErr, "page not found")
```